blog/_posts/06-03-blogpost.md

```diff
@@ -0,0 +1,164 @@
+
+---
+layout: post
+title: "BitVector Optimizations and DataMap Features"
+date: 2020-06-03
+author: Jamie Schmidt
+---
+# BitVector Optimizations and DataMap Functionality Improvements
+
+## Introduction 
+For my project, I worked with the Empirical library.  I primarily focused on BitVector and DataMap, two classes for data storage.  
+
+BitVector is, as the name suggests, a Vector of bits (though it's actually bytes).  The standard library has something similar, std::BitMap, but it is incapable of bit operations like NOT or XOR and has a size which must be fixed at compile time.  BitVector, on the other hand, has bit operation support and can be resized during runtime.  However, BitVector did not contain what is known as a "small optimization", that being an optimization where a class has all of its data stored in its own memory, without relying on a pointer elsewhere.   
+
+DataMap is similar to a regular map, but is heavily optimized for scientific computing.  All of its memory is stored in a single array of bytes, and layout information (detailing what is stored where) can be shared between separate DataMaps.  However, DataMap did not contain a way to store a reference to another object in the same place as the object itself.  In other words, given a variable in a DataMap, you couldn't set that variable to be a reference elsewhere.  When you added that variable, you had to choose whether it was going to be the variable itself or a pointer to that variable.  
```

Comment:
In terms of markdown readability, I think the standard is to have each sentence on its own line